src/drivers/samv7/tone_alarm/CMakeLists.txt

```diff
@@ -35,7 +35,8 @@ px4_add_module(
 	MODULE drivers__tone_alarm
 	MAIN tone_alarm
 	SRCS
-		tone_alarm.cpp
+		tone_alarm_samv7.cpp
+		../../tone_alarm/tone_alarm.cpp
```

Comment:
We should do this by creating separate libraries in each folder and linking.
To link all you have to do is add the library in the DEPENDS of a px4 module.